src/pass/llvmgen.cpp

```diff
@@ -362,7 +371,10 @@ Value* LLVMGen::visit(FetchDataPtr& fetch_data_ptr)
     auto data_addr = builder()->CreateGEP(lltype(fetch_data_ptr.type.deref()),
                                           buf_addr, idx_val);
 
-    return data_addr;
+    Instruction* data_instr = cast<Instruction>(data_addr);
+    data_instr->setMetadata(LLVMContext::MD_noalias,
+                            MDNode::get(llctx(), ArrayRef<Metadata*>()));
+    return data_instr;
```

Comment:
are we sure we need to set alias in all these places?

Reply:
also is it possible to move this logic to a function, instead of repeating in all these places

Reply:
> are we sure we need to set alias in all these places?

It's recommended to put on function args & returns, and load & store instructions. It's less common to add to the CreateGEP or CreateAlloca but I saw a couple of examples that did it, so figured I'd add it in just in case. But I can take it out for sake of readability since those two probably don't make a difference.

Reply:
> also is it possible to move this logic to a function, instead of repeating in all these places

Sure, I can make a function that does this for the Loads and Stores since those are repeated the most.

Reply:
Updated